src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+import fs from 'fs';
+import commander from 'commander';
+import { PolykeyAgent, PeerInfo } from '../lib/Polykey';
+import {} from '../../tmp/hole-punch-test/publicserver';
+import { actionRunner, pkLogger, PKMessageType, determineNodePath } from '.';
+
+function makeAddPeerCommand() {
+  return new commander.Command('add')
+    .description('add a new peer to the store')
+    .option('-b, --base64 <base64>', 'decode the peer info from a base64 string')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-a, --connected-address <connectedAddress>', 'public address on which the node can be contacted')
+    .option('-r, --relay-key <relayKey>', 'path to the file which contains the public key of the relay peer')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const base64String = options?.base64?.replace('\r', '')?.replace('\n', '');
+
+        let successful: boolean;
+        if (base64String != undefined) {
```

Comment:
This condition should be made more clearer. What's the reason for this arbitrary base64 string?

Reply:
this is is because there is no type checking in commander options. You need to check whether it was provided but we can do that above.